core/templates/dev/head/components/promo/PromoBarDirective.js

```diff
@@ -19,25 +19,33 @@
  */
 
 oppia.directive('promoBar', [
-  'UrlInterpolationService', function(UrlInterpolationService) {
+  'PromoBarService', 'UrlInterpolationService', 'ENABLE_PROMO_BAR',
+  function(PromoBarService, UrlInterpolationService, ENABLE_PROMO_BAR) {
     return {
       restrict: 'E',
-      scope: {
-        getPromoMessage: '&promoMessage'
-      },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/promo/' +
         'promo_bar_directive.html'),
       controller: [
-        '$scope',
-        function($scope) {
+        '$scope', '$window',
+        function($scope, $window) {
           var isPromoDismissed = function() {
             return !!angular.fromJson(sessionStorage.promoIsDismissed);
           };
           var setPromoDismissed = function(promoIsDismissed) {
             sessionStorage.promoIsDismissed = angular.toJson(promoIsDismissed);
           };
 
+          if (ENABLE_PROMO_BAR) {
+            PromoBarService.getPromoBarData().then(function(promoBarObject) {
+              $scope.promoBarIsEnabled = promoBarObject.promoBarEnabled;
+              $scope.promoBarMessage = promoBarObject.promoBarMessage;
+            });
+          } else {
+            $scope.promoBarIsEnabled = false;
```

Comment:
Set these first at the outset (outside the if/else). Then, do the if statement and update them if necessary.

That way, the variables always have a value, and are never undefined.

Reply:
I think the logic should be in `PromoBarService` and here we should just call `PromoBarService.getPromoBarData()`

Reply:
Done.